dkc/core/migrations/0010_auto_20201023_0128.py

```diff
@@ -0,0 +1,35 @@
+from django.conf import settings
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ('core', '0009_merge_default_site'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='Collection',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+            ],
+        ),
+        migrations.RemoveField(
+            model_name='folder',
+            name='root_folder',
+        ),
+        migrations.AlterField(
+            model_name='file',
+            name='creator',
+            field=models.ForeignKey(editable=False, null=True, on_delete=django.db.models.deletion.SET_NULL, to=settings.AUTH_USER_MODEL),
+        ),
+        migrations.AddField(
+            model_name='folder',
+            name='collection',
+            field=models.ForeignKey(default=None, on_delete=django.db.models.deletion.CASCADE, related_name='all_folders', to='core.collection'),
```

Comment:
Note that this default is actually invalid. We can address in 2 ways:
* Squash all the migrations, and require that existing databases be dropped.
* Create a 3-step migration:
  * Add the field, but make it nullable
  * `RunPython`, where we can traverse in a more controlled ordering and populate the field
  * Alter the field, making it non-nullable; the default specified here should never actually be used

Reply:
I'm a fan of dropping existing dev databases. I've already ended up doing it several times.

Reply:
Will do.